File: javascript/grammer/15_bisect_left.js
function bisectLeft(arr, target) {
  let left = 0;
  let right = arr.length;
  while (left < right) {
    const mid = Math.floor((left + right) / 2);
    if (arr[mid] < target) left = mid + 1;
    else right = mid;
  }
  return left;
}

function bisectRight(arr, target) {
  let left = 0;
  let right = arr.length;
  while (left < right) {
    const mid = Math.floor((left + right) / 2);
    if (arr[mid] <= target) left = mid + 1;
    else right = mid;
  }
  return left;
}

const arr = [1, 2, 2, 2, 5];
const target = 2;
const left = bisectLeft(arr, target);
const right = bisectRight(arr, target);

console.log('배열:', arr);
console.log('target의 왼쪽 경계:', left);
console.log('target 개수:', right - left);

File: python/grammer/15_bisect_left.py
"""15) bisect_left: 정렬 배열에서 삽입 위치 찾기."""
from bisect import bisect_left, bisect_right

arr = [1, 2, 2, 2, 5]
target = 2

left = bisect_left(arr, target)
right = bisect_right(arr, target)

print("배열:", arr)
print("target의 왼쪽 경계:", left)
print("target 개수:", right - left)
